extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java

```diff
@@ -195,7 +198,10 @@ private ImageReference getImageReference(ContainerImageConfig containerImageConf
             throw new IllegalArgumentException("The supplied container-image repository '" + repository + "' is invalid");
         }
 
-        String tag = containerImageConfig.tag.orElse(applicationInfo.getVersion());
+        List<String> tags = new ArrayList<>();
+        containerImageConfig.tag.ifPresent(tags::add);
+        tags.addAll(containerImageConfig.tags.orElse(Collections.singletonList(applicationInfo.getVersion())));
```

Comment:
Shouldn't this be `containerImage.getAdditonalTags()` considering we have most of the work done https://github.com/quarkusio/quarkus/pull/9040/files#diff-11cc649f490ec2882648b7efcf89760aR20-R22 and it seems like we are repeating ourselves?

Reply:
Well, I wanted to, but at this point, `additionalTags` may be empty. While building `ContainerImageInfoBuildItem`, I use the first tag to build the image name which will be used by `docker build` command or jib build process. 

The tag could be extracted from `ContainerImageInfoBuildItem.getImage()` by splitting on `:`but I'm not sure this is the cleanest way to do it. Maybe, I should keep the image name and the tags separated. WDYT?

Reply:
> Well, I wanted to, but at this point, `additionalTags` may be empty. While building `ContainerImageInfoBuildItem`, I use the first tag to build the image name which will be used by `docker build` command or jib build process.

The way build steps works is that this method https://github.com/quarkusio/quarkus/pull/9040/files#diff-780e4883fba9afa8b58c67fe3f4d07fdR93 will not be called unless otherwise all the required build items are there. 
And from the looks of it (https://github.com/quarkusio/quarkus/pull/9040/files#diff-11cc649f490ec2882648b7efcf89760aR20-R22), `additionalTags` will at least contain something. Or I am missing something?

Reply:
Yes it will contains something at this point. But when building `ContainerImageInfoBuildItem` (https://github.com/quarkusio/quarkus/pull/9040/files#diff-cb1fafe1b90298f02ae4cd91f4120c18R36-R41), I use the first tag as a `default` one and remove it from `additionalTags`. Thus, the injected `ContainerImageInfoBuildItem.getAdditionalTag` may be empty as the first tag has been removed. I don't know if I'm right or if I miss something but I agree about removing this duplicated code.

Reply:
Yes it will contains something at this point. But when building ContainerImageInfoBuildItem (https://github.com/quarkusio/quarkus/pull/9040/files#diff-cb1fafe1b90298f02ae4cd91f4120c18R36-R41), I use the first tag as a default one (this is the one used to build the container) and remove it from additionalTags. Thus, the injected ContainerImageInfoBuildItem.getAdditionalTag may be empty as the first tag has been removed. I don't know if I'm right or if I miss something but I agree about removing this duplicated code.

Reply:
@machi1990, were you able to have a look?

Reply:
Hi @glefloch sorry I missed your ping. 

> Thus, the injected ContainerImageInfoBuildItem.getAdditionalTag may be empty as the first tag has been removed. 

I do not understand the need to add and remove a tag.

I guess what I was really hoping to see is having the tag handling logic somewhere centralised hiding the bits needed to handle the deprecated `quarkus.container-image.tag`.  Which will also prove easier to logic in the future once the said property is removed. 

Probably having the container info like `name` and `tags` in the `ContainerImageInfoBuildItem` would be easier and instead of building the `image` in constructor, you could have methods like:
 - `.getTags()` and `.getName()` (if needed), 
- `.getBaseImage()` that returns the image with first tag (zero indexed tag)
- `.getAdditionalmages()` - returning rest of images using tag with index > 0 which could be used in `DockerProcessor`
- `.getAdditionalTags()`  returning tags with index > 0  used in `JibProcessor` 
- etc

Probably @geoand has better ideas, so let's wait for his call and focus on getting the tests to pass till then.

Reply:
Thanks @machi1990, I understood. At this point, all tags can be verify. I moved all tags related logic into `ContainerImageInfoBuildItem`. I also fixed tests. There is still something weird when `quarkus.container-image.tag` is missing. I will push once everything is ok

Reply:
@glefloch Please ping me when you are done.